packages/@aws-cdk/aws-route53/lib/records/_util.ts

```diff
@@ -15,12 +15,15 @@ import { HostedZoneRef } from '../hosted-zone-ref';
  *        <li>Otherwise, append +.+, +zoneName+ and a trailing +.+</li>
  *      </ul>
  */
-export function determineFullyQualifiedDomainName(providedName: string, hostedZone: HostedZoneRef): string {
+export function determineFullyQualifiedDomainName(providedName: string, hostedZone: HostedZoneRef | string): string {
```

Comment:
Is this worth a type union? It could equally well be two functions, and that seems more readable.